### PR TITLE
CMake and MacOSX

### PR DESCRIPTION
**Pull request by yageek**
I just add small elements to make the compilation worked on OS X and makes the created project runnning.

### CODE REVIEW

File: templates/cpp-template-default/CMakeLists.txt [hunk 1]
```diff
@@ -114,6 +128,7 @@ include_directories(
   ${COCOS2D_ROOT}/cocos/platform
   ${COCOS2D_ROOT}/cocos/audio/include/
   Classes
+  /usr/local/include/entityx
```
Comment (zilongshanren) [inline templates/cpp-template-default/CMakeLists.txt]: We don't need to include `/usr/local/include/entityx`?

**Comment by zilongshanren** [comment]
@yageek  :+1:  The Mac Cmake support is broken for a long time.
Thanks for the fix.

File: templates/cpp-template-default/CMakeLists.txt (hunk 1 above)
Comment (yageek) [inline templates/cpp-template-default/CMakeLists.txt]: Of course. I fix it today. I will also try to clean a little bit

**Comment by pandamicro** [comment]
[ci rebuild]

**Comment by pandamicro** [comment]
@yangws can we merge this PR ?

File: templates/cpp-template-default/CMakeLists.txt [hunk 2]
```diff
@@ -134,21 +148,50 @@ set(GAME_HEADERS
...
+      PROPERTIES
+      MACOSX_PACKAGE_LOCATION Resources/fonts
+      )
   else()
     add_executable(${APP_NAME} ${GAME_SRC})
   endif ( WIN32 )
 endif()
 
-target_link_libraries(${APP_NAME} cocos2d)
+target_link_libraries(${APP_NAME} cocos2d ${PLATFORM_SPECIFIC_LINK_LIBRARIES})
```
Comment (yangws) [inline templates/cpp-template-default/CMakeLists.txt:194]: Is there any case let the variable PLATFORM_SPECIFIC_LINK_LIBRARIES be empty?
If the variable PLATFORM_SPECIFIC_LINK_LIBRARIES is empty, target_link_libraries command could failed, isn't it?

**Comment by ShuraLiu** [comment]
@yangws @pandamicro I download the v3.9beta0 and find that this PR have not merged into it. When can you finish it?

**Comment by minggo** [comment]
Cmake on Mac work ok now. And this PR has conflict, so i close this PR.

**Pull request closed without merging**